pkg/app/retry.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
```

Comment:
I initially called this `retry` but then I cant have a struct with the same word as the prefix (linter).

Reply:
`app` is kind of vague. I think id rather see this structure:

```
cmd
  retry
    main.go
  webhook
    main.go

pkg
  retry
    server.go
  webhook
    server.go
```

You can call your struct Server since the package is retry.

Reply:
This will require some refactoring of existing code...

Reply:
ack

Reply:
For consistency Im going to rename `GitHubMetricsAggregatorServer` to `Server` because it will be in the newly renamed webhook package.